wa/workloads/motionmark/__init__.py

```diff
@@ -0,0 +1,77 @@
+#    Copyright 2014-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import re
+
+from wa import UiautoWorkload, Parameter
+from wa.framework.exception import ValidationError, WorkloadError
+from wa.utils.types import list_of_strs
+from wa.utils.misc import unique
+
+
+class Motionmark(UiautoWorkload):
+
+    name = 'motionmark'
+    regex_matches = [re.compile(r'Multiply Score (.+)'),
+                     re.compile(r'Canvas Score (.+)'),
+                     re.compile(r'Leaves Score (.+)'),
+                     re.compile(r'Paths Score (.+)'),
+                     re.compile(r'Canvas Lines Score (.+)'),
+                     re.compile(r'Focus Score (.+)'),
+                     re.compile(r'Images Score (.+)'),
+                     re.compile(r'Design Score (.+)'),
+                     re.compile(r'Suits Score (.+)')]
+    score_regex = re.compile(r'.*?([\d.]+).*')
+    description = '''
+    A workload to execute the motionmark web based benchmark
+
+    Test description:
+    1. Open browser application
+    2. Navigate to the motionmark website - http://browserbench.org/motionmark/
+    3. Execute the benchmark
+
+    '''
+
+    def __init__(self, target, **kwargs):
+        super(Motionmark, self).__init__(target, **kwargs)
+        self.gui.timeout = 1500
+
+    def setup(self, context):
+        super(Motionmark, self).setup(context)
+        url = 'am start -a android.intent.action.VIEW -d https://browserbench.org/MotionMark/'
+        self.target.execute(url)
```

Comment:
Why not use `self.gui.start_activity()` here?

Reply:
Hi Sergei, 

I'm wondering what is this method? How does it work and what is the benefit over what I have?

Reply:
The `gui` object encapsulates all GUI interactions, so the workload itself does not issue specific GUI commands, and instead deals with the abstracted "gui".

The benefit is that all GUI interactions are preformed in a "standard" manner. If something needs to be updated (e.g. to handle a future android version) it would need to be updated in only one place. This also introduces consistency across different workloads' code.

Reply:
The `start_activity` method is actually part of the `PackageHandler` rather than `UiAutomatorGUI` (`gui`). In this case it is relying on a pre-installed browser to execute the action rather than installing one and therefore wouldn't have access to that method. We do however have an `open_url` method available as part `AndroidTarget` which could be used instead.